### PR TITLE
Started on Flag

### PR DESCRIPTION
**Pull request by skumbananaaa**
## Purpose
  - This has to do with the "Win/Lose due to Game objective" user story.

## Changes
 - Some changes had to be made to the ResourceLoader. Previously, we pretransformed all vertices into world space. This is not a good solution because it makes it hard to manipulate objects in the scene, as they are not in local space. This has now been reworked so that we load the vertices in local space and instead calculate a default position, rotation and scale of each mesh. These values are then saved in the Mesh object and the SpecialObject.
 - No new classes or files were added, except for the "CollisionGroups.h" file in CrazyCanvas. This file defines values for collision groups only used in CrazyCanvas. Most other changes are in ResourceLoader and LevelObjectCreator.

### CODE REVIEW

File: CrazyCanvas/Include/Physics/PhysicsGroups.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
```
Comment (TheoBerlin) [inline CrazyCanvas/Include/Physics/PhysicsGroups.h]: Perhaps `CollisionGroups.h` is a more suitable name, to stay consistent with the name of the enum.

File: CrazyCanvas/Source/World/LevelObjectCreator.cpp [hunk 2]
```diff
@@ -195,11 +200,74 @@ ESpecialObjectType LevelObjectCreator::CreateSpawnpoint(const LambdaEngine::Spec
...
+	PhysicsSystem* pPhysicsSystem	= PhysicsSystem::GetInstance();
+
+	Entity entity = pECS->CreateEntity();
+
+	pECS->AddComponent<OffsetComponent>(entity, OffsetComponent{ .Offset = glm::vec3(0.0f)});
+	pECS->AddComponent<ParentComponent>(entity, ParentComponent{ .Attached = false });
+	const CollisionCreateInfo collisionCreateInfo =
+	{
+		.Entity				= entity,
+		.Position			= pECS->AddComponent<PositionComponent>(entity,		{ true, specialObject.DefaultPosition/* + translation + glm::vec3(0.0f, specialObject.DefaultScale.y * specialObject.BoundingBoxes[0].HalfExtent.y, 0.0f)*/}),
```
Comment (TheoBerlin) [inline CrazyCanvas/Source/World/LevelObjectCreator.cpp]: Did you forget to remove a temporary comment?

File: CrazyCanvas/Source/World/LevelObjectCreator.cpp [hunk 3]
```diff
@@ -195,11 +200,74 @@ ESpecialObjectType LevelObjectCreator::CreateSpawnpoint(const LambdaEngine::Spec
...
+			//Set Flag Carrier (Parent)
+			flagParentComponent.Attached	= true;
+			flagParentComponent.Parent		= playerEntity;
+
+			//Set Flag Offset
+			const Mesh* pMesh = ResourceManager::GetMesh(playerMeshComponent.MeshGUID);
+			flagOffsetComponent.Offset		= glm::vec3(0.0f, pMesh->BoundingBox.Dimensions.y / 2.0f, 0.0f);
+
+			Level::OnFlagPickedUp();
+		}
```
Comment (TheoBerlin) [inline CrazyCanvas/Source/World/LevelObjectCreator.cpp]: Two things about this callback function:
#### Where to place the callback function definition
I think the definition of the callback function should be somewhere else. I think it's best to have it as a member function of some class handling gameplay logic.

#### The callback function should schedule a job
The callback function makes accesses to components when other systems and jobs might be running. To avoid data races and other threading errors, such accesses should be made within a job. A job consists of two things:
1. An array of component types accessed during the job's execution, and with what permissions (read or read-write).
2. A function to execute

TLDR; create a job within the callback function, and schedule it using `ECSCore::ScheduleJobASAP(...) `or `ECSCore::ScheduleJobPostFrame(...)`. The latter executes after all systems have finished.

File: CrazyCanvas/engine_config_sandbox.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
   "FixedTimestep": 60,
   "NetworkPort": 4444,
   "RayTracingEnabled": true,
-  "MeshShadersEnabled": false,
+  "MeshShadersEnabled": true,
```
Comment (TheoBerlin) [inline CrazyCanvas/engine_config_sandbox.json]: Is this intentional?

**Review by Mumsfilibaba — APPROVED** [review]
Looks good. Maybe there is a better solution for the parentMatrix but I do not really see a reason for it more than it looks better (And a bit safer 👀) nor do I have a better solution.

**Review by TheoBerlin — CHANGES_REQUESTED** [review]
༼  ͡° ͜ʖ ͡° ༽

File: CrazyCanvas/Include/Physics/PhysicsGroups.h (hunk 1 above)
Comment (skumbananaaa) [reply CrazyCanvas/Include/Physics/PhysicsGroups.h]: Yup. my mind must have been somewhere else

File: CrazyCanvas/Source/World/LevelObjectCreator.cpp (hunk 2 above)
Comment (skumbananaaa) [reply CrazyCanvas/Source/World/LevelObjectCreator.cpp]: Yes, thanks!

File: CrazyCanvas/Source/World/LevelObjectCreator.cpp (hunk 3 above)
Comment (skumbananaaa) [reply CrazyCanvas/Source/World/LevelObjectCreator.cpp]: The job thing I agree with, but the callback function placement I don't really agree with.

My reasoning behind this is that I would like `LevelObjectCreator ` to be the main place where we define all Level Entity functionality, mainly to avoid jumping between so many different files when changing stuff, which we have to do for some other things right now and I find it annoying.

File: CrazyCanvas/engine_config_sandbox.json (hunk 4 above)
Comment (skumbananaaa) [reply CrazyCanvas/engine_config_sandbox.json]: Yes, since the CrazyCanvas project uses MeshShaders I think Sandbox should use it too. They will automatically be disabled anyway if they aren't supported.

**Review by TheoBerlin — APPROVED** [review]
